packages/contracts-periphery/contracts/L2/L2ERC721Bridge.sol

```diff
@@ -115,16 +102,18 @@ contract L2ERC721Bridge is ERC721Bridge, Semver {
         address _to,
         uint256 _tokenId
     ) external onlySelf {
+        require(_localToken != address(this), "L2ERC721Bridge: local token cannot be self");
+
         require(
-            // slither-disable-next-line reentrancy-events
+            // Note that supportsInterface makes a callback to the _localToken address
+            // which is user provided.
             ERC165Checker.supportsInterface(_localToken, type(IOptimismMintableERC721).interfaceId),
             "L2ERC721Bridge: local token interface is not compliant"
         );
         require(
             _remoteToken == IOptimismMintableERC721(_localToken).remoteToken(),
             "L2ERC721Bridge: wrong remote token for Optimism Mintable ERC721 local token"
         );
-        require(_localToken != address(this), "L2ERC721Bridge: local token cannot be self");
 
```

Comment:
apologies for the slight larger diff, I also reordered the checks here to be more consistent between the L1 and L2 bridges.